components/tidb_qe/src/batch/executors/index_scan_executor.rs

```diff
@@ -2,28 +2,30 @@
 
 use std::sync::Arc;
 
-use cop_datatype::EvalType;
 use kvproto::coprocessor::KeyRange;
+use tidb_qe_datatype::EvalType;
 use tipb::executor::IndexScan;
 use tipb::expression::FieldType;
 use tipb::schema::ColumnInfo;
 
 use super::util::scan_executor::*;
-use crate::coprocessor::codec::batch::{LazyBatchColumn, LazyBatchColumnVec};
-use crate::coprocessor::dag::batch::interface::*;
-use crate::coprocessor::dag::expr::{EvalConfig, EvalContext};
-use crate::coprocessor::dag::storage::Storage;
-use crate::coprocessor::{Error, Result};
+use crate::batch::interface::*;
+use crate::codec::batch::{LazyBatchColumn, LazyBatchColumnVec};
+use crate::expr::{EvalConfig, EvalContext};
+use crate::storage::Storage;
+use crate::Result;
 
 pub struct BatchIndexScanExecutor<S: Storage>(ScanExecutor<S, IndexScanExecutorImpl>);
 
-impl<S: Storage> BatchIndexScanExecutor<S> {
+impl BatchIndexScanExecutor<Box<dyn Storage<Statistics = ()>>> {
```

Comment:
This `dyn Storage<Statistics = ()>` looks weird. Firstly I thought the `Storage` is be converted to a trait object somewhere, but finally find out it's just used to hide the type.
I think it deserves a comment or just use a free function instead.